include/zephyr/drivers/i2c/target/eeprom.h

```diff
@@ -21,6 +21,27 @@
  * @{
  */
 
+/**
+ * @brief Store data callback handler
+ *
+ * When this callback is present, it will get called when a write to the
+ * EEPROM is completed allowing a user to store the data permanently.
```

Comment:
I would actually drop this from the documentation of the callback and rather move it to `eeprom_target_set_store_data_cb` to better explain what it is about, since right now its documentation is pretty dry?